src/main/java/com/redhat/devtools/intellij/common/utils/ExecHelper.java

```diff
@@ -498,6 +500,36 @@ public static void executeWithTerminal(Project project, String title, String...
     executeWithTerminal(project, title, new File(HOME_FOLDER), true, Collections.emptyMap(), command);
   }
 
+  public static void executeWithTerminalWidget(Project project, String title, String... command) throws IOException {
+    executeWithTerminalWidgetInternal(project, HOME_FOLDER, title, command);
+  }
+
+  public static void executeWithTerminalWidgetInternal(Project project, String workingDirectory, String title, String... command) throws IOException {
+    ensureTerminalWindowsIsOpened(project);
+
+    final TerminalView view = TerminalView.getInstance(project);
+    final Method[] method = new Method[1];
+    final Object[][] parameters = new Object[1][];
+    try {
+      method[0] = TerminalView.class.getMethod("createLocalShellWidget", new Class[] {String.class, String.class});
+      parameters[0] = new Object[] { workingDirectory, title };
+    } catch (NoSuchMethodException e) {
+      try {
+        method[0] = TerminalView.class.getMethod("createLocalShellWidget", new Class[] {String.class});
+        parameters[0] = new Object[] { workingDirectory };
+      } catch (NoSuchMethodException e1) {
+        throw new IOException(e1);
+      }
+    }
```

Comment:
Invoking `createLocalShellWidget` via reflection (apparently in order to catch both API variants) could be extracted to a separate method for improved clarity/isolation.

Reply:
done this in my added commit